analyzers/src/SonarAnalyzer.CSharp/Rules/ForLoopCounterChanged.cs

```diff
@@ -107,9 +109,11 @@ private static IEnumerable<ISymbol> LoopCounters(ForStatementSyntax node, Semant
             return declaredVariables.Union(initializedVariables);
         }
 
-        private static IEnumerable<SyntaxNode> AffectedExpressions(SyntaxNode node) =>
-            node.DescendantNodesAndSelf()
-                .Where(n => SideEffectExpressions.Any(s => s.Kinds.Any(n.IsKind)))
-                .Select(n => SideEffectExpressions.Single(s => s.Kinds.Any(n.IsKind)).AffectedExpression(n));
+        private static SyntaxNode[] ComputeAffectedExpressions(SyntaxNode node) =>
+            (from descendantNode in node.DescendantNodesAndSelf()
+             from sideEffect in SideEffectExpressions
+             where descendantNode.IsAnyKind(sideEffect.Kinds.ToImmutableHashSet())
```

Comment:
```suggestion
             where descendantNode.IsAnyKind(sideEffect.Kinds)
```

Reply:
There's no overload of `IsAnyKind` that accepts an `ImmutableList` as input.
I think I'll change everything to `ImmutableSet`.